platform/FileHandle.cpp

```diff
@@ -35,7 +35,18 @@ off_t FileHandle::size()
 
 std::FILE *fdopen(FileHandle *fh, const char *mode)
```

Comment:
Do you think it would be more appropriate to move `fdopen` to mbed_retarget.cpp?

Reply:
I don't quite know how it got here. Somehow during review of `FileHandle` work, I think.

I think the logic was that nothing in `mbed_retarget.h` referenced `FileHandle`. That's no longer true. So moving it would make sense. Putting the two declarations together in the header would also be clearer. Will do that.

Another thing that would make sense is that we could now split that `mbed_retarget.cpp` into two - the bulk of it is now "POSIX -> FileHandle/directory", which is a separate job from the C library retargetting. They're no longer inherently coupled. The C library retargetting just uses the POSIX API.

Reply:
> Another thing that would make sense is that we could now split that mbed_retarget.cpp into two - the bulk of it is now "POSIX -> FileHandle/directory", which is a separate job from the C library retargetting. They're no longer inherently coupled. The C library retargetting just uses the POSIX API.

might be good to split it

Reply:
Main thing holding me back on doing it was that it would probably render the GitHub diff useless on this PR. Would it make sense as a separate "no-functional change" follow-up PR?